src/lambda/commands/deploySamApplication.ts

```diff
@@ -186,35 +204,104 @@ async function buildPackageDeploy(params: {
             params.invoker,
             params.channelLogger.logger,
         )
+    } catch (err) {
+        // Handle sam deploy Errors to supplement the error message prior to writing it out
+        const error = err as Error
+
+        params.channelLogger.logger.error(error)
+
+        const errorMessage = enhanceAwsCloudFormationInstructions(String(err), params.deployParameters)
+        params.channelLogger.channel.appendLine(errorMessage)
+
+        throw new Error('Deploy failed')
+    }
+}
+
+async function deploy(params: {
+    deployParameters: DeploySamApplicationParameters,
+    invoker: SamCliProcessInvoker,
+    channelLogger: ChannelLogger,
+    window: WindowFunctions,
+}): Promise<void> {
+    try {
+        params.channelLogger.channel.show(true)
+        params.channelLogger.info(
+            'AWS.samcli.deploy.workflow.start',
+            'Starting SAM Application deployment...'
+        )
+
+        await buildOperation(params)
+        await packageOperation(params)
+        await deployOperation(params)
 
-        const msg = localize(
+        params.channelLogger.info(
             'AWS.samcli.deploy.workflow.success',
             'Successfully deployed SAM Application to CloudFormation Stack: {0} with profile: {1}',
             params.deployParameters.destinationStackName,
             params.deployParameters.profile
         )
-        params.channelLogger.channel.appendLine(msg)
-        // TODO: Is this the right way to provide this feedback?
-        vscode.window.showInformationMessage(msg)
+
+        params.window.showInformationMessage(localize(
```

Comment:
Do we need two alerts?

Reply:
The output channel can be thought of as a more verbose accounting of what's going on.
Because deploys take some amount of time, users have likely moved on to do other things, and the showInfo is a toast to inform that deploy has completed. (The toast is standard across JB/VS toolkits)

Here's a sample of the two (remember that users may not have the Output pane open if they've moved on to do other things while deploying)

![image](https://user-images.githubusercontent.com/39839589/58276032-4844d300-7d4b-11e9-9515-2770e790d589.png)

Reply:
I agree that we should avoid spamming the user with alerts. But as Chris mentioned, in this case the operation can take quite some time, so I think that a toast makes sense in this case.

Reply:
Yeah, now that I see it in context, it should be fine.